### PR TITLE
Fix issues detected by phpstan at level 1

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | probably
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

First batch of fixes. It represents about 30% of detected issues.
Each commit could be reviewed separately, and we can move some of them in dedicated PR if it requires some rework.

### CODE REVIEW

File: src/User.php [hunk 1]
```diff
@@ -4438,17 +4438,15 @@ public static function dropdown($options = [])
         }
 
         // Make a select box with all glpi users
+        $view_users = self::canView();
+
         if (!$p['multiple']) {
             $user = getUserName($p['value'], 2, true);
-        }
```
Comment (orthagh) [inline src/User.php:4443]: i can't understand why the variable was undefined, and also, your change removed a closing brace, no ?

Comment (cedric-anne) [reply src/User.php:4443]: `$user` is only defined in `if (!$p['multiple']) {` block, so I move its usage in this block. Then, as there was 2 `if (!$p['multiple']) {` blocks following each other, I just merged them.

File: src/Ticket_Contract.php [hunk 2]
```diff
@@ -74,17 +74,17 @@ public static function displayTabContentForItem(
         $tabnum = 1,
         $withtemplate = 0
     ) {
-        if (!($item instanceof CommonDBTM)) {
```
[the remark this answers is not in the document]
Comment (cedric-anne) [reply src/Ticket_Contract.php:77]: Replaced by c3f77e6b9be7975626de9be913d4b29350a716e5

File: src/Ticket.php [hunk 3]
```diff
@@ -512,40 +512,29 @@ public function getDatasToAddOLA($olas_id, $entities_id, $date, $type)
...
-                $prefix        = "sla";
-                $prefix_ticket = "";
-                $level_ticket  = new SlaLevel_Ticket();
-                break;
-            case "OLA":
-                $prefix        = "ola";
-                $prefix_ticket = "internal_";
-                $level_ticket  = new OlaLevel_Ticket();
-                break;
+        if (!is_a($laType, LevelAgreement::class, true)) {
```
[the remark this answers is not in the document]
Comment (cedric-anne) [reply src/Ticket.php]: Replaced by 3da472470de8acb36c04e9870c1fc8c49b2924dc

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**